modules/internal/ArrayViewSlice.chpl

```diff
@@ -78,6 +78,50 @@ module ArrayViewSlice {
                       doiBulkTransferFromAny,  doiBulkTransferToAny;
 
 
+    //
+    // A helper routine to encode logic about when slices should be RVF'd.
+    // Currently we RVF slices whose domains and arrays are privatized
+    // (since doing so involves RVFing their PIDs) and that support the
+    // chpl__serialize() routine (and, we assume, chpl__deserialize())
+    //
+    proc chpl__rvfMe() param {
+      use Reflection;
+      if (dom.dsiSupportsPrivatization() && arr.dsiSupportsPrivatization() &&
+          canResolveMethod(dom, "chpl__serialize") && 
+          canResolveMethod(arr, "chpl__serialize")) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    //
+    // Serialize a slice (when chpl__rvfMe() says to) by serializing its
+    // domain and array
+    //
+    proc chpl__serialize() where chpl__rvfMe() {
```

Comment:
You might want to disable forwarding for chpl__serialize/deserialize for this ArrayView as well because it can forward if the where-clause is 'false'.